src/Uno.UI.DualScreen/DuoApplicationViewSpanningRect.cs

```diff
@@ -31,91 +32,140 @@ public IReadOnlyList<Rect> GetSpanningRects()
-							if (bounds.X == 0 && bounds.Y == 0)
+							if (occludedRects.Length > 1 && this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Warning))
 							{
-								var statusBarRect = StatusBar.GetForCurrentView().OccludedRect.LogicalToPhysicalPixels();
-								occludedRect.Y -= statusBarRect.Height;
+								this.Log().Warn($"DualMode: Unknown screen layout, more than one occluded region. Only first will be considered. Please report your device to Uno Platform!");
 							}
-						}
 
-						if (intersection != Rect.Empty)
-						{
-							if (occludedRect.X == bounds.X)
-							{
-								_spanningRects = new List<Rect> {
-										new Rect(bounds.X, bounds.Y, bounds.Width, occludedRect.Y),
-										new Rect(bounds.X, bounds.Y + occludedRect.Y + occludedRect.Height, bounds.Width, bounds.Height - (occludedRect.Y + occludedRect.Height)),
-									};
+							var bounds = wuxWindowBounds;
+							var occludedRect = occludedRects[0];
+							var intersection = bounds.IntersectWith(occludedRect);
 
-								if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+							if (wuOrientation == DisplayOrientations.Portrait || wuOrientation == DisplayOrientations.PortraitFlipped)
+							{
+								// Compensate for the status bar size (the occluded area is rooted on the screen size, whereas
+								// wuxWindowBoundsis rooted on the visible size of the window, unless the status bar is translucent.
+								if ((int)bounds.X == 0 && (int)bounds.Y == 0)
 								{
-									this.Log().Debug($"DualMode: Horizontal spanning rects: {string.Join(";", _spanningRects)}");
+									var statusBarRect = StatusBar.GetForCurrentView().OccludedRect.LogicalToPhysicalPixels();
+									occludedRect.Y -= statusBarRect.Height;
 								}
 							}
-							else if (occludedRect.Y == bounds.Y)
+
+							if (intersection != null) // Occluded region overlaps the app
 							{
-								// Horizontal
+								if ((int)occludedRect.X == (int)bounds.X)
+								{
+									// Vertical stacking
+									// +---------+
+									// |         |
+									// |         |
+									// +---------+
+									// +---------+
+									// |         |
+									// |         |
+									// +---------+
+
+									var spanningRects = new List<Rect> {
+										// top region
+										new Rect(bounds.X, bounds.Y, bounds.Width, occludedRect.Top),
+										// bottom region
+										new Rect(bounds.X,
+											occludedRect.Bottom,
+											bounds.Width,
+											bounds.Height - occludedRect.Bottom),
+									};
 
-								_spanningRects = new List<Rect> {
+									if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+									{
+										this.Log().Debug($"DualMode: Horizontal spanning rects: {string.Join(";", spanningRects)}");
+									}
+
+									_previousMode.result = spanningRects;
+								}
+								else if ((int)occludedRect.Y == (int)bounds.Y)
+								{
+									// Horizontal side-by-side
+									// +-----+ +-----+
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4993239)